### PR TITLE
@tarun/feature/typography button and vitest

### PR DESCRIPTION
**Pull request by TarunKumardev**
created two components button and typography and added configuration for testing library

### CODE REVIEW

File: src/stories/button/button.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  component: Button,
+};
+export default meta;
+type Story = StoryObj<typeof Button>;
+export const Block: Story = {
+  args: {
+    children: 'button',
+    variant: 'contained',
+  },
+};
```
Comment (Sahara24) [inline src/stories/button/button.stories.tsx:14]: Show all the variants that are there in the component, not only contained.

File: src/stories/button/index.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        {applyStartIcon}
+        {children}
+        {applyendIcon}
+      </a>
+    );
+  }
+
+  return (
+    <button
+      disabled={loading}
```
Comment (Sahara24) [inline src/stories/button/index.tsx]: Do we have to disable a button with loading, what if we want to show a loader inside a button ? Isn't it good to separate both?

File: src/stories/typography/typography.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+type Story = StoryObj<typeof Typography>;
+
+export const Block: Story = {
+  args: {
+    children: 'Hello world',
+    variant: 'heading1',
+    fontWeight: 'bolder',
+  },
+};
```
Comment (Sahara24) [inline src/stories/typography/typography.stories.tsx:18]: write stories for all the types available

**Comment by Sahara24** [comment]
Going forward make separate commits for each feature that you work, dont push all the features in one commit

File: src/stories/button/index.tsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  startIcon,
+  endIcon,
+  variant,
+  className,
+  href,
+  target,
+  onClick,
+  style,
+  ...rest
+}: IButton): ReactNode {
```
Comment (Sahara24) [inline src/stories/button/index.tsx:20]: Add hover effect also for button and make button similar to the UI in nike website

**Review by Sahara24 — CHANGES_REQUESTED** [review]
For button component add types, I mostly see black and white button with hover, take a look at it and make similar to those buttons, and I have added some comments look into it.

File: src/stories/button/button.stories.tsx (hunk 1 above)
Comment (TarunKumardev) [reply src/stories/button/button.stories.tsx:14]: @Sahara24  planning to create color schema component doing r&d on it  after completion of that component will integrate the colors

**Review by Sahara24 — APPROVED** [review]

**Review by amarnathbarpanda — APPROVED** [review]

**Pull request merged**